source/module_hsolver/genelpa/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+#cmake_minimum_required(VERSION 2.8)
+#PROJECT (TEST)
+
+#SET(ELPA_INCLUDE_DIR /home/nic/yshen/program/elpa/2016.05.004/include/elpa-2016.05.004/)
+#SET(ELPA_INCLUDE_DIR /home/nic/yshen/program/elpa/2021.11.002-A100/include/elpa-2021.11.002)
+MESSAGE(STATUS "elpa include dir:" ${ELPA_INCLUDE_DIR} )
+string(REGEX REPLACE "/$" "" ELPA_INC_STR ${ELPA_INCLUDE_DIR})
+string(REPLACE "/" ";" ELPA_INC_STR ${ELPA_INC_STR})
+list(GET ELPA_INC_STR -1 ELPA_VER)
+string(REPLACE "elpa-" "" ELPA_VER ${ELPA_VER})
+message(STATUS "elpa version is: " ${ELPA_VER})
```

Comment:
Is it possible to get the elpa version from header file, like [modules/FindELPA.cmake](https://github.com/abacusmodeling/abacus-develop/pull/171/files#diff-b99d60b24a7770488bf35977dfad7f653142eeabd7964896f3ba5361ca97a76a)?

Reply:
> in https://github.com/abacusmodeling/abacus-develop/pull/171/files/4774d4492d2ec192f807ca034ced9d0a2ae6b8dd#diff-6d9fd3d243dafc589abb0ed35cbc82bb2b9f36ade9fc2ec91956c45ab8fff267

This method extracts elpa version info from include file path. However, those information won't be always available. The elpa installed from package management system does not contain version in folder name, e.g. `/usr/include/elpa/elpa/elpa.h`.
Personally, I think this is a dirty workaround. The version of elpa should be retrieved from `elpa_version.h`, but I'm not 100% sure if the legacy elpa contains this file.